index.js

```diff
@@ -197,14 +244,14 @@ function convertReactComponentFunction(path, context) {
   });
 
   let name = '';
-  if (
-    path.type === 'ArrowFunctionExpression' &&
-    path.parent &&
-    path.parent.type === 'VariableDeclarator'
-  ) {
-    name = path.parent.id.name;
-  } else if (path.type === 'FunctionDeclaration' && path.node.id && path.node.id.name) {
+  if (path.type === 'FunctionDeclaration' && path.node.id && path.node.id.name) {
```

Comment:
For function declarations, it's easy to resolve the name since there are no variable declarators or anything but for function expressions, it's parent will be a declarator and the code before checked that but with forwardRef, memo and memo(forwardRef) the declarator won't be the direct parent, it'll be the parent's parent for forwardRef and memo or for memo(forwardRef) it'll be the parent of the parent's parent.

Rather than checking those three cases, I used findParent to find the declarator.